### DISCUSSION TITLE
Express-based REST API on Vercel?

### DISCUSSION THREAD

**Discussion by sahithyandev** [vercel/vercel#4418] [Help] [answered]
What do you think about hosting an Express-based REST API on Vercel?
Please tell with the reason.

**Comment by mcsdevv** [comment, marked as answer]
Hello @KSahithyan, you are more than welcome to use the Express package, however, the Vercel platform is serverless and is a fundamentally different model to your typical 'Express API'.

The platform is being suited for frontends with added Serverless Functions, you can find more information about those here: https://vercel.com/docs/v2/serverless-functions/introduction

**Comment by sahithyandev** [comment]
Thank you @mcsdevv.
But What would be the case if I use Express API on Vercel? 
Because I love Vercel. I want to deploy my API on my favorite hosting service.

https://tv-shows-api-psi.now.sh/

**Reply by paulogdm** [reply]
It works, but the platform is simply not the best fit for it. A serverful environment for Express APIs may be better, such as Heroku, Digital Ocean, etc.

**Reply by rfgamaral** [reply]
Hi @paulogdm, would love to understand a bit more why a platform like Vercel is not the best fit for a REST API. What are the downsides of using Vercel to host a REST API? What can't be done with Vercel that is achievable through Heroku, for instance?

**Reply by wi-ski** [reply]
The 'surface level' reasons - afaik - focus on general statefulness. Ie. CRUD operations leveraging stateful database connections.  It appears many people simply ignore that however.
